include/grpc++/impl/call.h

```diff
@@ -172,17 +172,31 @@ class CallOpRecvMessage {
   grpc_byte_buffer* recv_buf_;
 };
 
+namespace CallOpGenericRecvMessageHelper {
+class DeserializeFunc {
+ public:
+  virtual Status deser(grpc_byte_buffer* buf,int max_message_size) = 0;
+};
+
+template<class R> class DeserializeFuncType : public DeserializeFunc {
+ public:
+  DeserializeFuncType(R *message): message_(message) {}
+  Status deser(grpc_byte_buffer* buf,int max_message_size) {
+    return SerializationTraits<R>::Deserialize(buf, message_,
+					       max_message_size);
+  }
+ private:
+  R *message_; // Not a managed pointer because management is external to this
+};
+}; // namespace CallOpGenericRecvMessageHelper
+
 class CallOpGenericRecvMessage {
  public:
   CallOpGenericRecvMessage() : got_message(false) {}
 
-  template <class R>
-  void RecvMessage(R* message) {
-    deserialize_ = [message](grpc_byte_buffer* buf,
-                             int max_message_size) -> Status {
-      return SerializationTraits<R>::Deserialize(buf, message,
-                                                 max_message_size);
-    };
+  template <class R> void RecvMessage(R* message) {
+    deserialize_.reset(new CallOpGenericRecvMessageHelper::
```

Comment:
It's exchanging one allocation for another, so no more concerning than what we had.

(Note that std::function<> will allocate something to hold the lambda, since lambda's can vary in size).